### PR TITLE
move builtin chains to yaml file

### PR DESCRIPTION
**Pull request by boojamya**
## Description

This PR moves the builtin chains from a hardcoded struct to a yaml file.

This makes it straight forward for testers to view and edit the "preconfigured" settings.

Ibctest will embed ./confiugredChains.yaml. It will refer to this embedded version UNLESS the environment variable  IBCTEST_CONFIGURED_CHAINS is set to a path for "custom" preconfigured chains and settings.

This should satisfy: https://github.com/strangelove-ventures/ibctest/issues/175

### CODE REVIEW

File: chainfactory.go [hunk 1]
```diff
@@ -42,16 +45,40 @@ type BuiltinChainFactory struct {
...
+// initBuiltinChainConfig returns an ibc.ChainConfig mapping all configured chains
+func initBuiltinChainConfig() (map[string]ibc.ChainConfig, error) {
+
+	// checks if configuredChains.yaml is found in cwd or parent of cwd,
+	// otherwise it will use the the embedded version of configuredChains.yaml
+	configuredChainsFile := ""
+	if _, err := os.Stat("./configuredChains.yaml"); err == nil {
+		configuredChainsFile = "./configuredChains.yaml"
+	}
+	if _, err := os.Stat("../configuredChains.yaml"); err == nil {
```
Comment (mark-rushakoff) [inline chainfactory.go:60]: I'm not so sure about this. I would typically be very surprised about defaulting to a file in a single parent directory.

File: chainfactory.go [hunk 2]
```diff
@@ -42,16 +45,40 @@ type BuiltinChainFactory struct {
...
+	}
+
+	builtinChainConfigs := make(map[string]ibc.ChainConfig)
+
+	var dat []byte
+	var err error
+	if configuredChainsFile != "" {
+		dat, err = os.ReadFile(configuredChainsFile)
+		if err != nil {
+			return nil, fmt.Errorf("error reading configuredChains.yaml: %v", err)
```
Comment (mark-rushakoff) [inline chainfactory.go:71]: This should use the actual path of the file.

File: chainspec.go [hunk 3]
```diff
@@ -75,6 +75,11 @@ func (s *ChainSpec) Config() (*ibc.ChainConfig, error) {
 		return s.applyConfigOverrides(s.ChainConfig)
 	}
 
+	builtinChainConfigs, err := initBuiltinChainConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get preconfigured chains")
```
Comment (mark-rushakoff) [inline chainspec.go:80]: Wrap the actual returned error.

File: configuredChains.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
+## NOTICE: This file gets embedded into ibctest binary
```
Comment (mark-rushakoff) [inline configuredChains.yaml:1]: Should this file be generated from another source, or do we have to hardcode its contents?

File: ibc/types.go [hunk 5]
```diff
@@ -9,27 +9,27 @@ import (
...
+	ChainID string `yaml:"chain-id"`
 	// Docker images required for running chain nodes.
-	Images []DockerImage
+	Images []DockerImage `yaml:"images"`
 	// Binary to execute for the chain node daemon.
-	Bin string
+	Bin string `yaml:"bin"`
 	// Bech32 prefix for chain addresses, e.g. cosmos.
-	Bech32Prefix string
+	Bech32Prefix string `yaml:"bech32Prefix"`
```
Comment (mark-rushakoff) [inline ibc/types.go:22]: All the other names are hyphenated except this one. That is confusing.

**Review by mark-rushakoff — COMMENTED** [review]
There are a couple things I'm not sure about.

File: chainfactory.go (hunk 1 above)
Comment (boojamya) [reply chainfactory.go:60]: As discussed on our call. Going to prioritize the embedded `configuredChains.yaml` unless an environment variable is present.

File: configuredChains.yaml (hunk 4 above)
Comment (boojamya) [reply configuredChains.yaml:1]: For lets "hardcode" this. But the idea around this PR is to make this info more transparent and easily editable.

**Review by mark-rushakoff — APPROVED** [review]
Rename the `once` variable, but otherwise this looks good to go.

**Pull request merged**